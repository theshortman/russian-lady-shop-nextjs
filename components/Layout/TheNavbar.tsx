import React, { FC } from "react";
import Link from "next/link";

const TheNavbar: FC = () => {
  const [showMobileMenu, setShowMobileMenu] = React.useState(false);

  return (
    <nav className="navbar navbar-expand-lg navbar-light border-bottom mb-4">
      <div className="container-fluid">
        <Link href="/">
          <a className="navbar-brand">Русская Леди</a>
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavAltMarkup"
          aria-controls="navbarNavAltMarkup"
          aria-expanded={showMobileMenu}
          aria-label="Toggle navigation"
          onClick={() => setShowMobileMenu(!showMobileMenu)}
        >
          <span className="navbar-toggler-icon"></span>
        </button>

        <div
          id="navbarNavAltMarkup"
          className={`collapse navbar-collapse ${showMobileMenu ? "show" : ""}`}
        >
          <div className="navbar-nav ms-auto">
            <Link aria-current="page" href="/shipping-and-payment">
              <a className="nav-link">Доставка и Оплата</a>
            </Link>
            <Link href="/size-chart">
              <a className="nav-link">Таблица размеров</a>
            </Link>
            <Link href="/contacts">
              <a className="nav-link">Контакты</a>
            </Link>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default TheNavbar;
