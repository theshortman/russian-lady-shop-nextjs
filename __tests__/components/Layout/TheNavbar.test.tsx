import { render } from "@testing-library/react";
import TheNavbar from "@/components/Layout/TheNavbar";

test("renders TheNavbar unchanged", () => {
  const { container } = render(<TheNavbar />);
  expect(container).toMatchSnapshot();
});
