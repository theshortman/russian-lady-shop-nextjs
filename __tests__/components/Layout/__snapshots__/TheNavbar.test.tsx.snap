// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TheNavbar unchanged 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light border-bottom mb-4"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Русская Леди
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse "
        id="navbarNavAltMarkup"
      >
        <div
          class="navbar-nav ms-auto"
        >
          <a
            class="nav-link"
            href="/shipping-and-payment"
          >
            Доставка и Оплата
          </a>
          <a
            class="nav-link"
            href="/size-chart"
          >
            Таблица размеров
          </a>
          <a
            class="nav-link"
            href="/contacts"
          >
            Контакты
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
